bundles/org.openhab.binding.casokitchen/src/main/java/org/openhab/binding/casokitchen/internal/handler/TwoZonesWinecoolerHandler.java

```diff
@@ -0,0 +1,255 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.casokitchen.internal.handler;
+
+import static org.openhab.binding.casokitchen.internal.CasoKitchenBindingConstants.*;
+
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.openhab.binding.casokitchen.internal.config.TwoZonesWinecoolerConfiguration;
+import org.openhab.binding.casokitchen.internal.dto.LightRequest;
+import org.openhab.binding.casokitchen.internal.dto.StatusRequest;
+import org.openhab.binding.casokitchen.internal.dto.StatusResult;
+import org.openhab.core.i18n.TimeZoneProvider;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TwoZonesWinecoolerHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class TwoZonesWinecoolerHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(TwoZonesWinecoolerHandler.class);
+
+    private final TimeZoneProvider timeZoneProvider;
+    private final HttpClient httpClient;
+    private Optional<ScheduledFuture<?>> refreshJob = Optional.empty();
+    private Optional<StatusResult> cachedResult = Optional.empty();
+    private TwoZonesWinecoolerConfiguration configuration = new TwoZonesWinecoolerConfiguration();
+
+    public TwoZonesWinecoolerHandler(Thing thing, HttpClient hc, TimeZoneProvider tzp) {
+        super(thing);
+        httpClient = hc;
+        timeZoneProvider = tzp;
+    }
+
+    @Override
+    public void initialize() {
+        configuration = getConfigAs(TwoZonesWinecoolerConfiguration.class);
+        if (checkConfig()) {
+            startSchedule();
+        }
+    }
+
+    private boolean checkConfig() {
```

Comment:
Within the flow it works as expected, but io would like to ask you to refactor this. The method is not only checking the config (as it's name suggests), it is also setting the Thing state. 

1. Maybe return the reason and set the thing state in the initialize method
2. Another fix could be to change the method name.
3. On top of the above i would propose to change the logich a bit:
 - make use of `isBlank()` instead of `!= EMPTY`
 - prevend nested if's. : check condition and exit /return immediate.

Reply:
Fixed similar to MercedesMe binding pattern